packages/headless/src/test/mock-engine.ts

```diff
@@ -72,6 +73,24 @@ interface MockEngine {
 
 type DispatchExts = ThunkDispatch<AppState, void, AnyAction>;
 
+export interface MockSSRSearchEngine extends MockSearchEngine {
+  waitForSearchCompletedAction(): Promise<SearchCompletedAction>;
+}
+
+/**
+ * For internal use only.
+ *
+ * Returns a non-functional `SSRSearchEngine`.
```

Comment:
Btw, this kind of JSDoc for internal mock function is not necessary/a bit overkill 😄

Reply:
Yea I just matched the style of the other mockEngine functions. They all have a doc description !